src/engine/sidechain/shoutconnection.cpp

```diff
@@ -881,53 +887,65 @@ void ShoutConnection::updateMetaData() {
+                kLogger.info()
+                        << "Updating static metadata for"
+                        << profile()->getProfileName()
+                        << ": song:" << songStr;
+                const QByteArray baCustomSong = encodeString(songStr);
                 insertMetaData("song", baCustomSong.constData());
             }
 
             setFunctionCode(13);
             shout_set_metadata(m_pShout, m_pShoutMetaData);
-            m_firstCall = true;
+            m_firstCall = false;
         }
     }
 }
 
 void ShoutConnection::errorDialog(const QString& text, const QString& detailedError) {
     qWarning() << "Streaming error: " << detailedError;
-    ErrorDialogProperties* props = ErrorDialogHandler::instance()->newDialogProperties();
-    props->setType(DLG_WARNING);
-    props->setTitle(tr("Connection error"));
-    props->setText(tr("One of the Live Broadcasting connections raised this error:<br>"
-            "<b>Error with connection '%1':</b><br>")
-            .arg(profile()->getProfileName()) + text);
-    props->setDetails(detailedError);
-    props->setKey(detailedError);   // To prevent multiple windows for the same error
-    props->setDefaultButton(QMessageBox::Close);
-    props->setModal(false);
-    ErrorDialogHandler::instance()->requestErrorDialog(props);
+    ErrorDialogProperties* pProps = ErrorDialogHandler::instance()->newDialogProperties();
+    pProps->setType(DLG_WARNING);
+    pProps->setTitle(tr("Connection error"));
+    pProps->setText(tr("One of the Live Broadcasting connections raised this error:<br>"
+                       "<b>Error with connection '%1':</b><br>")
+                            .arg(profile()->getProfileName()) +
+            text);
+    pProps->setDetails(detailedError);
+    pProps->setKey(detailedError); // To prevent multiple windows for the same error
+    pProps->setDefaultButton(QMessageBox::Close);
+    pProps->setModal(false);
+    ErrorDialogHandler::instance()->requestErrorDialog(pProps);
     setState(NETWORKSTREAMWORKER_STATE_ERROR);
 }
 
 void ShoutConnection::infoDialog(const QString& text, const QString& detailedInfo) {
-    ErrorDialogProperties* props = ErrorDialogHandler::instance()->newDialogProperties();
-    props->setType(DLG_INFO);
-    props->setTitle(tr("Connection message"));
-    props->setText(tr("<b>Message from Live Broadcasting connection '%1':</b><br>")
-            .arg(profile()->getProfileName()) + text);
-    props->setDetails(detailedInfo);
-    props->setKey(text + detailedInfo);
-    props->setDefaultButton(QMessageBox::Close);
-    props->setModal(false);
-    ErrorDialogHandler::instance()->requestErrorDialog(props);
+    ErrorDialogProperties* pProps = ErrorDialogHandler::instance()->newDialogProperties();
+    pProps->setType(DLG_INFO);
+    pProps->setTitle(tr("Connection message"));
+    pProps->setText(tr("<b>Message from Live Broadcasting connection '%1':</b><br>")
```

Comment:
That has probably been translated already, but I know I was told to keep formatting out of the `tr` string to prevent potential format corruption when translators update these. Shall we sanitise this string and extract the formatting? 

```suggestion
    pProps->setText(QStringLiteral("<b>")+tr("Message from Live Broadcasting connection '%1'")+QStringLiteral(":</b><br>"))
```

If you agree, there is one above to update too